### PR TITLE
Feature/vudrm 1421/geo whitelisting

### PR DESCRIPTION
**Pull request by PerritonBranch**

### CODE REVIEW

File: docs/source/VUDRM-token.md [hunk 1]
```diff
@@ -39,7 +39,8 @@ The DRM policy is included in the VUDRM token as the encrypted third component.
...
 | `duration_lease`   | string   | Seconds                   | Fairplay       | Overrides polend for Fairplay lease licenses.                                                              |
 | `duration_persist` | string   | Seconds                   | Fairplay       | Overrides polend for Fairplay persist licenses.                                                           |
-| `match_content_id` | bool     | `true` or `false`         | Fairplay, Widevine       | Boolean for if the content id in the token should be compared with the content id in the used in the license request; the content id in the license request will be the same as the content id used to generate the keys that encrypted a given piece of content.                                                     
+| `match_content_id` | bool     | `true` or `false`         | Fairplay, Widevine       | Boolean for if the content id in the token should be compared with the content id in the used in the license request; the content id in the license request will be the same as the content id used to generate the keys that encrypted a given piece of content.
```
Comment (fschurer) [inline docs/source/VUDRM-token.md:42]: Repetition of 'in the' ('...with the content id in the used in the licence request')

File: docs/source/VUDRM-token.md [hunk 2]
```diff
@@ -50,6 +51,9 @@ There are also limitations depending on environments that are not explained in t
 ### Match content id
 If `content_id` has been set and `match_content_id` has been set to `true`, when a license request is made the content id in the license request and the content id in the VUDRM token will be compared. If they **are** the same a license will be served as normal; if they are **not** the same the license request will be denied. If `match_content_id` has been set to `false` **or** `content_id` has not been specified then this comparision will not occur. 
 
+### Geo whitelisting
+If `geo_whitelist` has been added to the DRM policy, when a license is requested the geo location of the clients ip will be checked. If the country is in the whitelist the license request will be successful. If the country is not in the whitelist the request will be denied. E.g. if the DRM policy is `{ "geo_whitelist":[ "gbr", "deu" ] }` only users from the UK and Germany will be able to make successful license request.
```
Comment (fschurer) [inline docs/source/VUDRM-token.md:55]: Missing apostrophe (client's instead of clients).

File: docs/source/VUDRM-token.md [hunk 3]
```diff
@@ -39,7 +39,8 @@ The DRM policy is included in the VUDRM token as the encrypted third component.
...
 | `duration_persist` | string   | Seconds                   | Fairplay       | Overrides polend for Fairplay persist licenses.                                                           |
-| `match_content_id` | bool     | `true` or `false`         | Fairplay, Widevine       | Boolean for if the content id in the token should be compared with the content id in the used in the license request; the content id in the license request will be the same as the content id used to generate the keys that encrypted a given piece of content.                                                     
+| `match_content_id` | bool     | `true` or `false`         | Fairplay, Widevine       | Boolean for if the content id in the token should be compared with the content id used in the license request; the content id in the license request will be the same as the content id used to generate the keys that encrypted a given piece of content. 
+| `geo_whitelist`     | string array | Array of 3 letter country codes (ISO format) | Fairplay, Widevine | A list of 3 letter country codes, in ISO format, for all countries that are allowed access.
```
Comment (somerobmitchell) [inline docs/source/VUDRM-token.md:43]: needs iso version

File: docs/source/VUDRM-token.md [hunk 4]
```diff
@@ -50,6 +51,9 @@ There are also limitations depending on environments that are not explained in t
 ### Match content id
 If `content_id` has been set and `match_content_id` has been set to `true`, when a license request is made the content id in the license request and the content id in the VUDRM token will be compared. If they **are** the same a license will be served as normal; if they are **not** the same the license request will be denied. If `match_content_id` has been set to `false` **or** `content_id` has not been specified then this comparision will not occur. 
 
+### GEO whitelisting
+If `geo_whitelist` has been added to the DRM policy, when a license is requested the GEO location of the client's ip will be checked. If the country is in the whitelist the license request will be successful. If the country is not in the whitelist the request will be denied. E.g. if the DRM policy is `{ "geo_whitelist":[ "gbr", "deu" ] }` only users from the UK and Germany will be able to make successful license requests.
```
Comment (somerobmitchell) [inline docs/source/VUDRM-token.md:55]: needs iso version

**Comment by somerobmitchell** [comment]
do not merge added as needs to wait for code to be live.

**Pull request merged**